compute_tools/src/bin/compute_ctl.rs

```diff
@@ -256,6 +317,19 @@ fn main() -> Result<()> {
         }
     }
 
+    Ok(WaitSpecResult { compute, http_port })
+}
+
+struct WaitSpecResult {
+    compute: Arc<ComputeNode>,
+    // passed through from ProcessCliResult
+    http_port: u16,
+}
+
+fn start_postgres(
```

Comment:
This function contains quite a bit of things that are unrelated to starting Postgres. Could we chop it up as well?

Reply:
Same hesitancy here as with `cleanup_and_exit`. I'll see what I can do.

I think the "right" way to do this is probably to have some abstraction around starting/stopping vm-monitor stuff, so that the usage in `compute_ctl.rs` is just one-liners, but I considered that outside the scope of this PR.

Happy to include it as a separate commit if desired.

Reply:
Some of this will be handled by #7601 - other than that, I think it's just separating out the vm-monitor stuff, which I'm hesitant to do. Maybe better to name it something other than `start_postgres`... `start_compute_tasks` ? not sure. WDYT?